lib/sanbase/clickhouse/miners_balance.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Sanbase.Clickhouse.MinersBalance do
+  @moduledoc ~s"""
+  Uses ClickHouse to calculate miner balances over time.
+  """
+
+  alias Sanbase.DateTimeUtils
+  require Sanbase.ClickhouseRepo, as: ClickhouseRepo
+
+  @table "eth_miners_metrics"
+
+  @type balance :: %{datetime: DateTime.t(), balance: number()}
+
+  @spec balances(
+          String.t(),
+          DateTime.t(),
+          DateTime.t(),
+          String.t()
+        ) :: {:ok, list(balance)} | {:error, String.t()}
+  def balances("ethereum", from, to, interval) do
+    interval_in_seconds = DateTimeUtils.compound_duration_to_seconds(interval)
+
+    case rem(interval_in_seconds, 86_400) do
+      0 ->
+        calculate_balances(from, to, interval_in_seconds)
+
+      _ ->
+        {:error, "The interval must consist of whole days!"}
+    end
+  end
+
+  def balances(_, _, _, _), do: {:error, "Currently only ethereum is supported!"}
+
+  def first_datetime(_) do
+    query = """
+    SELECT
+      toUnixTimestamp(min(dt))
```

Comment:
```suggestion
      toUnixTimestamp(min(date))
```

`dt` is not a valid column here